src/browser/browser.ts

```diff
@@ -324,14 +324,22 @@ export class Browser {
         if (this.config.verboseLogging) {
           this.log.debug('Waiting for dashboard/panel to load', 'timeout', `${options.timeout}s`);
         }
+
         return page.waitForFunction(
-          () => {
+          (isFullPage) => {
+            if (isFullPage) {
+              const panelCount = document.querySelectorAll('[data-panelId]').length;
+              const totalPanelsRendered = document.querySelectorAll('.panel-content').length + document.querySelectorAll('.dashboard-row').length;
```

Comment:
its probably worth adding a comment what `.panel-content` refers to and why do we add the `.dashboard-row`